pkg/lifecycle/startup/startup.go

```diff
@@ -772,6 +777,51 @@ func autoDetectCIDRBySkipInterface(ifaceRegexes []string, version int) *cnet.IPN
 	return cidr
 }
 
+// autoDetectUsingK8sInternalIP reads K8s Node InternalIP.
+func autoDetectUsingK8sInternalIP(version int, nodeName string) *cnet.IPNet {
+	// If running under kubernetes with secrets to call k8s API
+	if config, err := rest.InClusterConfig(); err == nil {
```

Comment:
nit: flip the condition and return nil on error rather then indenting everything below?